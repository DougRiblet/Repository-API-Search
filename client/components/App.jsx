import React, { useState } from 'react';
import Search from './Search.jsx';
import Detail from './Detail.jsx';
import Results from './Results.jsx';
import searchGithubApi from '../utils/searchGithubApi.js';

export default function App() {
  const [showModal, setShowModal] = useState(false);
  const [results, setResults] = useState([]);
  const [error, setError] = useState('');
  const [detail, setDetail] = useState({});

  const fetchRepos = async (searchTerm, langFilter, sortOption) => {
    setError('');
    const newResults = await searchGithubApi(searchTerm, langFilter, sortOption);
    if (Array.isArray(newResults)) {
      setResults(newResults);
    } else {
      setResults([]);
      setError(newResults.error);
    }
  };

  const showDetail = (repoId) => {
    const selectedRepo = results.filter((el) => el.id === repoId)[0];
    setDetail(selectedRepo);
    setShowModal(true);
  };

  const handleCloseModal = () => {
    setShowModal(false);
    setDetail({});
  };

  return (
    <div>
      <main>
        <div>
          <Search
            fetchRepos={fetchRepos}
          />
          <Results
            results={results}
            showDetail={showDetail}
            error={error}
          />
          <Detail
            repo={detail}
            showModal={showModal}
            handleCloseModal={handleCloseModal}
          />
        </div>
      </main>
    </div>
  );
}
