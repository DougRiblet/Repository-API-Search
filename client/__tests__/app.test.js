import React from "react";
import { create } from "react-test-renderer";

import App from "../components/App.jsx";

describe("App Snapshot", () => {
  test("App component matches snapshot", () => {
    const appSnapshot = create(<App />);
    expect(appSnapshot.toJSON()).toMatchSnapshot();
  });
});
