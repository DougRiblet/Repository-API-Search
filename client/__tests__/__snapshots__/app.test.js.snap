// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Snapshot App component matches snapshot 1`] = `
<div>
  <main>
    <div>
      <div
        className="search-container"
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="search-form"
          >
            <div
              className="term"
            >
              <label
                htmlFor="search-text"
              >
                Search Term (required)
                <input
                  id="search-text"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              className="language"
            >
              <label
                htmlFor="lang-text"
              >
                Filter by Language (optional)
                <input
                  id="lang-text"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              className="sort"
            >
              <div
                className="sort-radio-header"
              >
                Sort Results By
              </div>
              <div
                className="radio-choices"
              >
                <div
                  className="radio"
                >
                  <label
                    htmlFor="default-radio"
                  >
                    <input
                      checked={true}
                      id="default-radio"
                      onChange={[Function]}
                      type="radio"
                      value="default"
                    />
                    Default
                  </label>
                </div>
                <div
                  className="radio"
                >
                  <label
                    htmlFor="stars-radio"
                  >
                    <input
                      checked={false}
                      id="stars-radio"
                      onChange={[Function]}
                      type="radio"
                      value="stars"
                    />
                    Stars
                  </label>
                </div>
              </div>
            </div>
            <div
              className="submit-button"
            >
              <input
                disabled={true}
                type="submit"
                value="Search Github"
              />
            </div>
          </div>
        </form>
      </div>
      <div
        className="results-container"
      >
        
        <div
          className="results-grid"
          role="grid"
        />
      </div>
      <div
        className="modal display-none"
        role="document"
      >
        <div
          className="modal-container"
        >
          <button
            id="close-modal-button"
            onClick={[Function]}
            type="button"
          >
            Back to Search Results
          </button>
          <div
            className="detail-display"
          >
            <h2 />
            <ul
              className="detail-list"
            >
              <li>
                owner: 
              </li>
              <li>
                language: 
              </li>
              <li>
                created: 
              </li>
              <li>
                updated: 
              </li>
              <li>
                open issues: 
              </li>
              <li>
                forks: 
              </li>
            </ul>
            <p>
              
            </p>
            <p>
              â˜… 
            </p>
            <p>
              <a>
                View Repo at Github
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
